### PR TITLE
Cleans up type conversions, adds CPU test comparing with NumPy

### PR DESCRIPTION
**Pull request by mruberry**
Per title. Follow-up to https://github.com/pytorch/pytorch/pull/35086.

### CODE REVIEW

**Comment by vadimkantorov** [comment]
Also https://github.com/pytorch/pytorch/issues/34300 is probably related

File: test/test_torch.py [hunk 1]
```diff
@@ -15200,6 +15200,27 @@ def test_float_to_int_undefined_conversion(self, device, dtype):
         t = torch.tensor((-3.40282e+38, 3.40282e+38), device=device, dtype=torch.float)
         self.assertEqual(t.to(dtype).dtype, dtype)
 
+    # Note: CUDA will fail this test on most dtypes, often dramatically.
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    @onlyCPU
+    @dtypes(torch.bool, torch.uint8, torch.int8, torch.int16, torch.int32, torch.int64)
+    def test_float_to_int_conversion_precision(self, device, dtype):
+        t = torch.tensor((-3.40282e+38, 3.40282e+38), device=device, dtype=torch.float)
+        a = np.array((-3.40282e+38, 3.40282e+38), dtype=np.float32)
```
Comment (ngimel) [inline test/test_torch.py]: `np.finfo(np.float32).max`? np.finfo is numpy for std::numeric_limits.

Comment (mruberry) [reply test/test_torch.py]: Sure.

**Review by ngimel — APPROVED** [review]
Looks good, thanks for removing intermediate cast.

**Comment by vadimkantorov** [comment]
Should pathological +-inf/nan cases also be compared to NumPy?

**Comment by mruberry** [comment]
> Should pathological +-inf/nan cases also be compared to NumPy?

Sure. I added them.

File: test/test_torch.py [hunk 2]
```diff
@@ -15200,6 +15200,29 @@ def test_float_to_int_undefined_conversion(self, device, dtype):
         t = torch.tensor((-3.40282e+38, 3.40282e+38), device=device, dtype=torch.float)
```
Comment (mruberry) [inline test/test_torch.py]: Use torch.finfo here

File: test/test_torch.py [hunk 3]
```diff
@@ -15200,6 +15200,29 @@ def test_float_to_int_undefined_conversion(self, device, dtype):
...
         self.assertEqual(t.to(dtype).dtype, dtype)
 
+    # Note: CUDA will fail this test on most dtypes, often dramatically.
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    @onlyCPU
+    @dtypes(torch.bool, torch.uint8, torch.int8, torch.int16, torch.int32, torch.int64)
+    def test_float_to_int_conversion_precision(self, device, dtype):
```
Comment (mruberry) [inline test/test_torch.py]: TODO (future): add more conversion tests

**Pull request closed without merging**